### PR TITLE
Updating registry with chain metadata

### PR DESCRIPTION
**Pull request by ekenigs**
This is a fix for issues https://github.com/polkadot-js/extension/issues/907 and https://github.com/polkadot-js/extension/issues/1061.

With updated metadata registry is providing a correct type for `ExtrinsicSignature` and the extension is signing transactions correctly.

### CODE REVIEW

File: packages/extension-base/src/background/handlers/Extension.ts [hunk 1]
```diff
@@ -232,6 +232,10 @@ export default class Extension {
     const { request, resolve } = queued;
 
     this.#state.saveMetadata(request);
+    request.metaCalls &&
```
Comment (jacogr) [inline packages/extension-base/src/background/handlers/Extension.ts:235]: This should not happen on save. It should kick in for the instance where the signing takes place. 

The issue you have here is that each update injects their own and they are different.

Comment (ekenigs) [reply packages/extension-base/src/background/handlers/Extension.ts]: Thanks, @jacogr  for your fast feedback. Moved to the signing method.

**Comment by ekenigs** [comment]
@jacogr hey, can you please check this?

**Comment by crystalin** [comment]
@jacogr gentle ping :)

**Review by JurbinM — APPROVED** [review]

**Comment by ekenigs** [comment]
@JurbinM hey, thanks for reviewing, can you please run checks and merge PR?

**Pull request closed without merging**